<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Benchmarking integrative multiomics methods for disease classification and biomarker discovery</title>
    <meta charset="utf-8" />
    <meta name="author" content="Tony Liang" />
    <script src="materials/libs/header-attrs/header-attrs.js"></script>
    <script src="materials/libs/htmlwidgets/htmlwidgets.js"></script>
    <link href="materials/libs/datatables-css/datatables-crosstalk.css" rel="stylesheet" />
    <script src="materials/libs/datatables-binding/datatables.js"></script>
    <script src="materials/libs/jquery/jquery-3.6.0.min.js"></script>
    <link href="materials/libs/dt-core/css/jquery.dataTables.min.css" rel="stylesheet" />
    <link href="materials/libs/dt-core/css/jquery.dataTables.extra.css" rel="stylesheet" />
    <script src="materials/libs/dt-core/js/jquery.dataTables.min.js"></script>
    <link href="materials/libs/crosstalk/css/crosstalk.min.css" rel="stylesheet" />
    <script src="materials/libs/crosstalk/js/crosstalk.min.js"></script>
    <script src="https://kit.fontawesome.com/ae71192e04.js" crossorigin="anonymous"></script>
    <link rel="stylesheet" href="materials/xaringan-themer.css" type="text/css" />
    <link rel="stylesheet" href="materials/slides-style.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">


layout: true

&lt;div class="my-footer"&gt;&lt;span&gt;&lt;a href="https://tonyliang19.github.io/ssrip_2023" style="color:white"&gt;tonyliang19.github.io/ssrip_2023&lt;/a&gt;&lt;/span&gt;&lt;/div&gt; 

---

background-image: url(materials/img/website_bkg.png)
background-position: top
background-size: contain


&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
.center[### Benchmarking integrative multiomics methods for disease classification and biomarker discovery]


.pull-left[
&lt;br/&gt;
####Tony Liang
####University of British Columbia
####2023-06-23
]

.pull-right[.center[.middle[
&lt;br/&gt;
&lt;table&gt;
&lt;tr&gt;
&lt;td&gt;![:scale 50%](materials/img/qr-slides.png)&lt;/td&gt;
&lt;td&gt;![:scale 50%](materials/img/qr-code.png)&lt;/td&gt;
&lt;td&gt;![:scale 50%](materials/img/qr-lab.png)&lt;/td&gt;
&lt;/tr&gt;
&lt;tr&gt;
&lt;td&gt;<svg aria-hidden="true" role="img" viewBox="0 0 640 512" style="height:1em;width:1.25em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M160 64c0-35.3 28.7-64 64-64H576c35.3 0 64 28.7 64 64V352c0 35.3-28.7 64-64 64H336.8c-11.8-25.5-29.9-47.5-52.4-64H384V320c0-17.7 14.3-32 32-32h64c17.7 0 32 14.3 32 32v32h64V64L224 64v49.1C205.2 102.2 183.3 96 160 96V64zm0 64a96 96 0 1 1 0 192 96 96 0 1 1 0-192zM133.3 352h53.3C260.3 352 320 411.7 320 485.3c0 14.7-11.9 26.7-26.7 26.7H26.7C11.9 512 0 500.1 0 485.3C0 411.7 59.7 352 133.3 352z"/></svg> Slides&lt;/td&gt;
&lt;td&gt;<svg aria-hidden="true" role="img" viewBox="0 0 640 512" style="height:1em;width:1.25em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M392.8 1.2c-17-4.9-34.7 5-39.6 22l-128 448c-4.9 17 5 34.7 22 39.6s34.7-5 39.6-22l128-448c4.9-17-5-34.7-22-39.6zm80.6 120.1c-12.5 12.5-12.5 32.8 0 45.3L562.7 256l-89.4 89.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0l112-112c12.5-12.5 12.5-32.8 0-45.3l-112-112c-12.5-12.5-32.8-12.5-45.3 0zm-306.7 0c-12.5-12.5-32.8-12.5-45.3 0l-112 112c-12.5 12.5-12.5 32.8 0 45.3l112 112c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L77.3 256l89.4-89.4c12.5-12.5 12.5-32.8 0-45.3z"/></svg> Code&lt;/td&gt;
&lt;td&gt;<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:currentColor;overflow:visible;position:relative;"><path d="M256 64C150 64 64 150 64 256s86 192 192 192c17.7 0 32 14.3 32 32s-14.3 32-32 32C114.6 512 0 397.4 0 256S114.6 0 256 0S512 114.6 512 256v32c0 53-43 96-96 96c-29.3 0-55.6-13.2-73.2-33.9C320 371.1 289.5 384 256 384c-70.7 0-128-57.3-128-128s57.3-128 128-128c27.9 0 53.7 8.9 74.7 24.1c5.7-5 13.1-8.1 21.3-8.1c17.7 0 32 14.3 32 32v80 32c0 17.7 14.3 32 32 32s32-14.3 32-32V256c0-106-86-192-192-192zm64 192a64 64 0 1 0 -128 0 64 64 0 1 0 128 0z"/></svg> Lab website&lt;/td&gt;
&lt;/tr&gt;
&lt;/table&gt;
]]]

???

* Explain a little in the QR
* Thanks to compbio lab to allow this project, everyone helped!


---







# Land acknowledgement

.pull-left[
**I would like to acknowledge that I work on the traditional, ancestral, and unceded territory of the Coast Salish Peoples, including the territories of the xwməθkwəy̓əm (Musqueam), Skwxwú7mesh (Squamish), Stó:lō and Səl̓ílwətaʔ/Selilwitulh (Tsleil- Waututh) Nations.**

*Traditional*: Traditionally used and/or occupied by Musqueam people

*Ancestral*: Recognizes land that is handed down from generation to generation

*Unceded*: Refers to land that was not turned over to the Crown (government) by a treaty or other agreement

]

.pull-right[
.center[&lt;img src ="materials/img/land_acknowledgement.gif" alt="matrix", width="80%"/&gt;]
]

---

## Outline

.pull-left[
.large[Background]
 
 - .secondary[Multiomics]
 
 - .secondary[Data integration]
 
&lt;!----
 - .secondary[Nextflow]
 - UBC ARC and Apptainer
 
----&gt;

.large[Motivation]

.large[Objectives]

.large[Methods]

.large[Expected Results]

.large[Conclusion ]

.large[Acknowledgements]

.large[.footnote[
For code to original repo see [.secondary[here]](https://github.com/tonyliang19/multiomics-pipeline)
]]
]

---

## Multiomics Data

Profiling of different molecular measurements (i.e. genes, proteins, metabolites) from same biological samples.


- Compensate for missing or unreliable information in any single .secondary[omics] data

- Multiple source of evidence pointing to same biological process are more likely


&lt;br&gt;

**Omics** is the comprehensive study of all moluecules of particular type.

- Each .secondary[omics] data follow `\(N \times P\)` structure
  + `\(N =\)` Samples of Observations
  
  + `\(P =\)` Predictors/variables
  
  + High dimensional, `\(N &lt;&lt; P\)`



---

### Multiomics Data Integration


&lt;br&gt;
&lt;br&gt;


.pull-left[

![](materials/img/multi-omics.png)

]

.pull-right[

![](materials/img/types-methods.png)

]

---

## Motivation

- Machine learning (`ML`) methods 

- Reproducible workflows and parallel computation

- Lack of CS knowledge in generic biological field

- "Advanced" data science

- **Multiomics** problem could also be viewed as mathematical or statistical problem

---

## Objectives

The objective is to benchmark methods for multiomics data integration by:

1. Curating publicly available multiomics data

2. Applying existing methods to simulated and real world datasets

3. Compare methods based on 
  + **classification accuracy **
  
  + .secondary[features selected]
  
  + `computation time`, etc

---

&lt;!----
background-image: url(materials/img/workflow.png)
background-size: contain
background-position: middle, center
-----&gt;

.center[![:scale 77%](materials/img/workflow.png)]

---

### Data

.pull-left-narrow[

- Simulated data 

- Bulk data (.secondary[Majority])

- Single-cell data

- Spatial data
]

.pull-right-wide[
![](materials/img/omics.png)
]


Many of these data are **high dimensional**, quite abstract to illustrate in plain words.


???

Briefly explain what each consits of

Bulk -&gt; majority of multiomics, this one ignores cell-type indenty, most basic one

Single-cell -&gt; new single cell technologies like CITE-seq , scRNA-seq, scATAC-seq and ECCITE-seq

Spaital -&gt; Spatial distirbution of cells is known

---

## Methods

.pull-left-narrow[
.secondary[Curate multiomics data from open-source databases
]

- Collect different omics data from same samples

- AnnData for `Python`

- MultiAssayExperiment for `R`

- Normalization , batch correction
]

.pull-right-wide[
.secondary[ `Benchmarking` integrative multiomics methods
]

- Evaluate series of methods based on their tasks with classic metrics

- Compare methods on different types of multiomics data

- Develop method that improve the predictive performance and
interpretability of the results.

- Use computational power and CS parallel algorithm

]

???

* Dont take too long in explain this
* Try making a 'consecutive' story

---

## Nextflow


.pull-left[
![:scale 97%](materials/img/nxf.png)
]
.pull-right[
![:scale 97%](materials/img/nextflow-story.png)
]


.pull-left[
- Highly customizable

- Parallel computation

- Portable in multiple platforms
]

.pull-right[

- Community pipelines and support

- Any programming Language

- <svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#e98a15;overflow:visible;position:relative;"><path d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg> Requires some CS knowledge

]

???

Only briefly mention the platform used and the bullets.
Rest could be explained in methods

---

class: center
&lt;!---
background-image: url(materials/img/arc.png)
background-size: 55%
----&gt;

### UBC Advanced Research Computing (ARC) Sockeye and Apptainer
&lt;br&gt;

.pull-left[
![:scale 90%](materials/img/arc.png)
]

&lt;br&gt;

.pull-right[
![:scale 100%](materials/img/apptainer.png)
]

???

Only briefly mention the platform
Rest could be explained in methods

---

## Expected Results

- Web portal app to quickly denote a method suitable to your custom data.

  .emphasis[Easily scalable to your own computing cluster]
  
- Time, error rates, number of features selected by method
  + Simulated (with/without noise)
  
  + Real dataset


- Availability of list of curated multiomics dataset
  
  + .secondary[Disease state]
  
  + .secondary[Number of samples/variables]
  
  + .secondary[Types of omics]
  
  + .secondary[Organ type]

---

## Expected Results

&lt;!--- Replace with other xlsx here from Amrit ---&gt;

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-e6f1485d7253fd94df00" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-e6f1485d7253fd94df00">{"x":{"filter":"none","vertical":false,"fillContainer":false,"data":[["1","2","3","4","5","6","7","8","9","10"],["Multiomics modeling of the immunome, transcriptome, microbiome, proteome and metabolome adaptations during human pregnancy","","","","","","","Cytokine-induced molecular responses in airway smooth muscle cells inform genome-wide assocation studies of asthma","",""],["longitudinal","","","","","","","case-control","",""],["plasma","plasma","serum","swab","blood","plasma","plasma","primary airway smooth muscle cells","primary airway smooth muscle cells","primary airway smooth muscle cells"],["cfRNA","proteins","proteins","microbiota","cells","metabolites","proteins","genotypes","mRNA","methylation"],["RNAseq","Luminex","Luminex","DNAseq","CyTOF","mass spec","Somalogic","Illumina Omni2.5v8v1A/Human Core Array","Illumina Human HT-12 v4 array","Illumina Human Methylation EPIC Beadchip"],[37275,62,62,18548,354,3485,1300,null,18279,786326],[68,null,null,null,null,null,null,268,null,null],["4 timepoints","","","","","","","asthma/non-asthma","",""],["multinomial","","","","","","","binary","",""],["","","","","","","","age, smoking status, race, sex, treatment","",""],["Bioinformatics 2019 35(1):95-103.","","","","","","","Genome Med. 2020 12(1):64.","",""],["https://nalab.stanford.edu/multiomics-pregnancy/","","","","","","","to ask authors","GSE146374","GSE146376"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Title<\/th>\n      <th>study design<\/th>\n      <th>sample-type<\/th>\n      <th>omic-type<\/th>\n      <th>technology<\/th>\n      <th>number of variables<\/th>\n      <th>number of observations<\/th>\n      <th>response<\/th>\n      <th>response-type<\/th>\n      <th>covariates<\/th>\n      <th>reference<\/th>\n      <th>data link/GEO<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":3,"scrollX":true,"searching":false,"columnDefs":[{"className":"dt-right","targets":[6,7]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[3,10,25,50,100],"rowCallback":"function(row, data, displayNum, displayIndex, dataIndex) {\nvar value=data[1]; $(this.api().cell(row, 1).node()).css({'font-size':'68%'});\nvar value=data[2]; $(this.api().cell(row, 2).node()).css({'font-size':'68%'});\nvar value=data[3]; $(this.api().cell(row, 3).node()).css({'font-size':'68%'});\nvar value=data[4]; $(this.api().cell(row, 4).node()).css({'font-size':'68%'});\nvar value=data[5]; $(this.api().cell(row, 5).node()).css({'font-size':'68%'});\nvar value=data[6]; $(this.api().cell(row, 6).node()).css({'font-size':'68%'});\nvar value=data[7]; $(this.api().cell(row, 7).node()).css({'font-size':'68%'});\nvar value=data[8]; $(this.api().cell(row, 8).node()).css({'font-size':'68%'});\nvar value=data[9]; $(this.api().cell(row, 9).node()).css({'font-size':'68%'});\nvar value=data[10]; $(this.api().cell(row, 10).node()).css({'font-size':'68%'});\nvar value=data[11]; $(this.api().cell(row, 11).node()).css({'font-size':'68%'});\nvar value=data[12]; $(this.api().cell(row, 12).node()).css({'font-size':'68%'});\n}"}},"evals":["options.rowCallback"],"jsHooks":[]}</script>


.large[
.footnote[
This is a yet to be updated data table (under curation)
]]

---

### Limitations and Future Direction

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#e98a15;overflow:visible;position:relative;"><path d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg> Collection and clean process of multiomics data

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#e98a15;overflow:visible;position:relative;"><path d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg> Missing source or supplementary materials from evaluated methods/papers

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#0a8754;overflow:visible;position:relative;"><path d="M323.8 34.8c-38.2-10.9-78.1 11.2-89 49.4l-5.7 20c-3.7 13-10.4 25-19.5 35l-51.3 56.4c-8.9 9.8-8.2 25 1.6 33.9s25 8.2 33.9-1.6l51.3-56.4c14.1-15.5 24.4-34 30.1-54.1l5.7-20c3.6-12.7 16.9-20.1 29.7-16.5s20.1 16.9 16.5 29.7l-5.7 20c-5.7 19.9-14.7 38.7-26.6 55.5c-5.2 7.3-5.8 16.9-1.7 24.9s12.3 13 21.3 13L448 224c8.8 0 16 7.2 16 16c0 6.8-4.3 12.7-10.4 15c-7.4 2.8-13 9-14.9 16.7s.1 15.8 5.3 21.7c2.5 2.8 4 6.5 4 10.6c0 7.8-5.6 14.3-13 15.7c-8.2 1.6-15.1 7.3-18 15.1s-1.6 16.7 3.6 23.3c2.1 2.7 3.4 6.1 3.4 9.9c0 6.7-4.2 12.6-10.2 14.9c-11.5 4.5-17.7 16.9-14.4 28.8c.4 1.3 .6 2.8 .6 4.3c0 8.8-7.2 16-16 16H286.5c-12.6 0-25-3.7-35.5-10.7l-61.7-41.1c-11-7.4-25.9-4.4-33.3 6.7s-4.4 25.9 6.7 33.3l61.7 41.1c18.4 12.3 40 18.8 62.1 18.8H384c34.7 0 62.9-27.6 64-62c14.6-11.7 24-29.7 24-50c0-4.5-.5-8.8-1.3-13c15.4-11.7 25.3-30.2 25.3-51c0-6.5-1-12.8-2.8-18.7C504.8 273.7 512 257.7 512 240c0-35.3-28.6-64-64-64l-92.3 0c4.7-10.4 8.7-21.2 11.8-32.2l5.7-20c10.9-38.2-11.2-78.1-49.4-89zM32 192c-17.7 0-32 14.3-32 32V448c0 17.7 14.3 32 32 32H96c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32H32z"/></svg> Feedback to bioinformatics or general research community

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#0a8754;overflow:visible;position:relative;"><path d="M323.8 34.8c-38.2-10.9-78.1 11.2-89 49.4l-5.7 20c-3.7 13-10.4 25-19.5 35l-51.3 56.4c-8.9 9.8-8.2 25 1.6 33.9s25 8.2 33.9-1.6l51.3-56.4c14.1-15.5 24.4-34 30.1-54.1l5.7-20c3.6-12.7 16.9-20.1 29.7-16.5s20.1 16.9 16.5 29.7l-5.7 20c-5.7 19.9-14.7 38.7-26.6 55.5c-5.2 7.3-5.8 16.9-1.7 24.9s12.3 13 21.3 13L448 224c8.8 0 16 7.2 16 16c0 6.8-4.3 12.7-10.4 15c-7.4 2.8-13 9-14.9 16.7s.1 15.8 5.3 21.7c2.5 2.8 4 6.5 4 10.6c0 7.8-5.6 14.3-13 15.7c-8.2 1.6-15.1 7.3-18 15.1s-1.6 16.7 3.6 23.3c2.1 2.7 3.4 6.1 3.4 9.9c0 6.7-4.2 12.6-10.2 14.9c-11.5 4.5-17.7 16.9-14.4 28.8c.4 1.3 .6 2.8 .6 4.3c0 8.8-7.2 16-16 16H286.5c-12.6 0-25-3.7-35.5-10.7l-61.7-41.1c-11-7.4-25.9-4.4-33.3 6.7s-4.4 25.9 6.7 33.3l61.7 41.1c18.4 12.3 40 18.8 62.1 18.8H384c34.7 0 62.9-27.6 64-62c14.6-11.7 24-29.7 24-50c0-4.5-.5-8.8-1.3-13c15.4-11.7 25.3-30.2 25.3-51c0-6.5-1-12.8-2.8-18.7C504.8 273.7 512 257.7 512 240c0-35.3-28.6-64-64-64l-92.3 0c4.7-10.4 8.7-21.2 11.8-32.2l5.7-20c10.9-38.2-11.2-78.1-49.4-89zM32 192c-17.7 0-32 14.3-32 32V448c0 17.7 14.3 32 32 32H96c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32H32z"/></svg> Extendable to other field that requires intensive computation

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#0a8754;overflow:visible;position:relative;"><path d="M323.8 34.8c-38.2-10.9-78.1 11.2-89 49.4l-5.7 20c-3.7 13-10.4 25-19.5 35l-51.3 56.4c-8.9 9.8-8.2 25 1.6 33.9s25 8.2 33.9-1.6l51.3-56.4c14.1-15.5 24.4-34 30.1-54.1l5.7-20c3.6-12.7 16.9-20.1 29.7-16.5s20.1 16.9 16.5 29.7l-5.7 20c-5.7 19.9-14.7 38.7-26.6 55.5c-5.2 7.3-5.8 16.9-1.7 24.9s12.3 13 21.3 13L448 224c8.8 0 16 7.2 16 16c0 6.8-4.3 12.7-10.4 15c-7.4 2.8-13 9-14.9 16.7s.1 15.8 5.3 21.7c2.5 2.8 4 6.5 4 10.6c0 7.8-5.6 14.3-13 15.7c-8.2 1.6-15.1 7.3-18 15.1s-1.6 16.7 3.6 23.3c2.1 2.7 3.4 6.1 3.4 9.9c0 6.7-4.2 12.6-10.2 14.9c-11.5 4.5-17.7 16.9-14.4 28.8c.4 1.3 .6 2.8 .6 4.3c0 8.8-7.2 16-16 16H286.5c-12.6 0-25-3.7-35.5-10.7l-61.7-41.1c-11-7.4-25.9-4.4-33.3 6.7s-4.4 25.9 6.7 33.3l61.7 41.1c18.4 12.3 40 18.8 62.1 18.8H384c34.7 0 62.9-27.6 64-62c14.6-11.7 24-29.7 24-50c0-4.5-.5-8.8-1.3-13c15.4-11.7 25.3-30.2 25.3-51c0-6.5-1-12.8-2.8-18.7C504.8 273.7 512 257.7 512 240c0-35.3-28.6-64-64-64l-92.3 0c4.7-10.4 8.7-21.2 11.8-32.2l5.7-20c10.9-38.2-11.2-78.1-49.4-89zM32 192c-17.7 0-32 14.3-32 32V448c0 17.7 14.3 32 32 32H96c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32H32z"/></svg> Save time from selecting reproducible ML methods for biological problem

.emphasis[

You don't need to understand tons of cs to use this project

]

---

## Conclusion

1. The best performing methods on the real-world data will provide insights into why certain methods are better than others with respect predictive performance and capturing the correct underlying biological processes. 

2. Inform the development of a new integrative method using .hand[logistic regression] that is applicable to different types of multiomics data.

3. Highly relevant to the community utilizing multiomics data for
their research.
  + Ease of use
  + Collaborations
  + Development of new tool

---

class: middle
background-image: url(materials/img/website_bkg.png)
background-position: top
background-size: contain

&lt;br&gt;
&lt;br&gt;
&lt;br&gt;

.large[.secondary[Thanks to my lab, the [CompBio Lab](https://cbl-hli.med.ubc.ca/)]]
.pull-left[
- Dr. Amrit Singh
- Dr. Dantong Zhu
- Dr. Young Woong Kim
- Dr. Maryam Ahmadzadeh
- Michael Yoon
]
.pull-right[
- Jeffrey Tang
- Rishika Daswani
- Roy He
- Samuel Leung
- Akshdeep Sandhu
- Asees Singh
]

&lt;br&gt;

And acknowledging to UBC ARC, work learn,  and Nextflow for enabling the successful execution of  this project.

.large[
.footnote[
For more information regarding Nextflow, please check [here](https://www.nextflow.io/)
]]



&lt;!----
Credits to:

  [Stat 406 Lecture Slides @DJM](https://ubc-stat.github.io/stat-406/)

&lt;br&gt;&lt;br&gt;
-----&gt;

    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script src="materials/macros.js"></script>
<script>var slideshow = remark.create({
"highlightLines": true,
"countIncrementalSlides": false,
"ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
