<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Benchmarking integrative multiomics methods for disease classification and biomarker discovery</title>
    <meta charset="utf-8" />
    <meta name="author" content="Tony Liang" />
    <script src="materials/libs/header-attrs/header-attrs.js"></script>
    <script src="materials/libs/htmlwidgets/htmlwidgets.js"></script>
    <link href="materials/libs/datatables-css/datatables-crosstalk.css" rel="stylesheet" />
    <script src="materials/libs/datatables-binding/datatables.js"></script>
    <script src="materials/libs/jquery/jquery-3.6.0.min.js"></script>
    <link href="materials/libs/dt-core/css/jquery.dataTables.min.css" rel="stylesheet" />
    <link href="materials/libs/dt-core/css/jquery.dataTables.extra.css" rel="stylesheet" />
    <script src="materials/libs/dt-core/js/jquery.dataTables.min.js"></script>
    <link href="materials/libs/crosstalk/css/crosstalk.min.css" rel="stylesheet" />
    <script src="materials/libs/crosstalk/js/crosstalk.min.js"></script>
    <script src="https://kit.fontawesome.com/ae71192e04.js" crossorigin="anonymous"></script>
    <link rel="stylesheet" href="materials/xaringan-themer.css" type="text/css" />
    <link rel="stylesheet" href="materials/slides-style.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">


layout: true

&lt;div class="my-footer"&gt;&lt;span&gt;&lt;a href="https://tonyliang19.github.io/ssrip_2023" style="color:white"&gt;tonyliang19.github.io/ssrip_2023&lt;/a&gt;&lt;/span&gt;&lt;/div&gt; 

---

background-image: url(materials/img/website_bkg.png)
background-position: top
background-size: contain


&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;&lt;br/&gt;
.center[### Benchmarking integrative multiomics methods for disease classification and biomarker discovery]


.pull-left[
&lt;br/&gt;
####Tony Liang
####University of British Columbia
####2023-06-23
]

---







# Land acknowledgement

.pull-left[
**I would like to acknowledge that I work on the traditional, ancestral, and unceded territory of the Coast Salish Peoples, including the territories of the xwməθkwəy̓əm (Musqueam), Skwxwú7mesh (Squamish), Stó:lō and Səl̓ílwətaʔ/Selilwitulh (Tsleil- Waututh) Nations.**

*Traditional*: Traditionally used and/or occupied by Musqueam people

*Ancestral*: Recognizes land that is handed down from generation to generation

*Unceded*: Refers to land that was not turned over to the Crown (government) by a treaty or other agreement

]

.pull-right[
.center[&lt;img src ="materials/img/land_acknowledgement.gif" alt="matrix", width="80%"/&gt;]
]

---

## Outline

.pull-left[
.large[Background]
 - .secondary[Multiomics data and integration]
 - .secondary[Nextflow]
 - UBC ARC and Apptainer
 

.large[Research Questions]

.large[Methods]

.large[Expected Results]

.large[Conclusion ]

.large[Acknowledgements]

.large[.footnote[
For code to original repo see [.secondary[here]](https://github.com/tonyliang19/multiomics-pipeline)
]]
]

.pull-right[
.center[&lt;img src ="materials/img/qr_to_repo.png" alt="matrix", width="80%"/&gt;]
]

???

* The QR code redirects to our lab website!

---

## Motivation

- Machine learning (`ML`) methods 

- Reproducible workflows and parallel computation

- Lack of CS knowledge in generic biological field

- "Advanced" data science

- **Multiomics** problem could also be viewed as mathematical or statistical problem




---

## Multiomics Data Integration

---

## Objectives

The objective is to benchmark methods for multiomics data integration by:

1. Curating publicly available multiomics data

2. Applying existing methods to simulated and real world datasets

3. Compare methods based on 
  + **classification accuracy **
  
  + .secondary[features selected]
  
  + `computation time`, etc

---

&lt;!----
background-image: url(materials/img/workflow.png)
background-size: contain
background-position: middle, center
-----&gt;

.center[![:scale 77%](materials/img/workflow.png)]

---

### Data

.pull-left-narrow[

- Simulated data 

- Bulk data (.secondary[Majority])

- Single-cell data

- Spatial data
]

.pull-right-wide[
![](materials/img/omics.png)
]


Many of these data are **high dimensional**, quite abstract to illustrate in plain words.


???

Briefly explain what each consits of

Bulk -&gt; majority of multiomics, this one ignores cell-type indenty, most basic one

Single-cell -&gt; new single cell technologies like CITE-seq , scRNA-seq, scATAC-seq and ECCITE-seq

Spaital -&gt; Spatial distirbution of cells is known

---

## Methods

.pull-left-narrow[
.secondary[Curate multiomics data from open-source databases
]

- Collect different omics data from same samples

- AnnData for `Python`

- MultiAssayExperiment for `R`

- Normalization , batch correction
]

.pull-right-wide[
.secondary[ `Benchmarking` integrative multiomics methods
]

- Evaluate series of methods based on their tasks with classic metrics

- Compare methods on different types of multiomics data

- Develop method that improve the predictive performance and
interpretability of the results.

- Use computational power and CS parallel algorithm

]

---

## Nextflow


.pull-left[
![:scale 97%](materials/img/nxf.png)
]
.pull-right[
![:scale 97%](materials/img/nextflow-story.png)
]


.pull-left[
- Highly customizable

- Parallel computation

- Portable in multiple platforms
]

.pull-right[

- Community pipelines and support

- Any programming Language

- <svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#e98a15;overflow:visible;position:relative;"><path d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg> Requires some CS knowledge

]

???

Only briefly mention the platform used and the bullets.
Rest could be explained in methods

---

class: center
&lt;!---
background-image: url(materials/img/arc.png)
background-size: 55%
----&gt;

### UBC Advanced Research Computing (ARC) Sockeye and Apptainer
&lt;br&gt;

.pull-left[
![:scale 90%](materials/img/arc.png)
]

&lt;br&gt;

.pull-right[
![:scale 100%](materials/img/apptainer.png)
]

???

Only briefly mention the platform
Rest could be explained in methods

---

## Expected Results

- Web portal app to quickly denote a method suitable to your custom data.

  .emphasis[Easily scalable to your own computing cluster]

- Availability of list of curated multiomics dataset
  
  + .secondary[Disease state]
  
  + .secondary[Number of samples/variables]
  
  + .secondary[Types of omics]
  
  + .secondary[Organ type]

---

## Expected Results

&lt;!--- Replace with other xlsx here from Amrit ---&gt;

<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-ec3ac46c6e86a2df523e" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-ec3ac46c6e86a2df523e">{"x":{"filter":"none","vertical":false,"fillContainer":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37"],["Alzheimer's Disease","","","","","","Triple Negative Breast Cancer","","","Type 2 Diabetes","","","Heart Failure","","","","","","","","","Schizophrenia","","","","","","IBDs","","","","","SLE","","","DLBCL",""],["Drugs","Drug Target Information","Genes","Expression Datasets","Metabolomic Datasets","SNPs","Gene Expression","Somatic Mutation Calling","Segmented Copy Number Data","DNA Methylation","GWAS","Gene Expression","Whole-exome sequencing data","Genotype data","RNA-seq data","Microarray data","DNA methylation data","Histone modification data","Mass spectrometry data","Immunoassay data","Metabolomics data","GWAS","RNA-seq data","DNA Methylation","CNV","Microarray data","SNPs","GWAS, CNV, rare variants","RNA-seq","DNA methylation and chromatin accessibility","Protein expression","Metabolite levels","Protein expression","Metabolite levels","Complex Lipid Levels","Mass Spectrometry Proteomics Data","RNA-seq data"],["13339","2892","41725","84","13","3322","Not clarified","","","254","9,000,000","x","&gt;1000","&gt;6000","&gt;1000","&gt;800","&gt;1000","&gt;800","&gt;800","&gt;3000","&gt;800","39","not clarified","not clarified","15","Not clarified","variable","&gt;8,000,000","2,500+","1,500+","500+","500+","777","1994","628","4774","17"],["&gt;100 datasets","","","","","","550","","","16","18801","63","13765","","","","","","","","","390,000","&lt;1000","&lt;1000","46000","&lt;1000","variable","3534","","","","","40","","","20",""],["https://alzgps.lerner.ccf.org.","","","","","","https://www.annalsofoncology.org/cms/10.1093/annonc/mdy024/attachment/a28fbae6-4519-4a13-90d0-9ee8a6ea470b/mmc1.zip","","","https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE21232","https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5034897/","https://www.ncbi.nlm.nih.gov/sites/GDSbrowser","http://heartfailure.medical-bigdata.com/Home","","","","","","","","","http://www.szdb.org/index.html","","","","","","https://ibdmdb.org","","","","","https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9063006/","","","PXD009089; ProteomeXchange Consortium Database","https://www.ebi.ac.uk/biostudies/arrayexpress/studies/E-MTAB-6597?query=E-MTAB-6597"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Disease/Physiological State<\/th>\n      <th>Datasets<\/th>\n      <th>Number of variables<\/th>\n      <th>Number of samples<\/th>\n      <th>Reference<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":5,"scrollX":true,"searching":false,"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false,"lengthMenu":[5,10,25,50,100]}},"evals":[],"jsHooks":[]}</script>


.large[
.footnote[
This is a yet to be updated data table (under curation)
]]

---

### Limitations and Future Direction

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#e98a15;overflow:visible;position:relative;"><path d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg> Collection and clean process of multiomics data

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#e98a15;overflow:visible;position:relative;"><path d="M459.1 52.4L442.6 6.5C440.7 2.6 436.5 0 432.1 0s-8.5 2.6-10.4 6.5L405.2 52.4l-46 16.8c-4.3 1.6-7.3 5.9-7.2 10.4c0 4.5 3 8.7 7.2 10.2l45.7 16.8 16.8 45.8c1.5 4.4 5.8 7.5 10.4 7.5s8.9-3.1 10.4-7.5l16.5-45.8 45.7-16.8c4.2-1.5 7.2-5.7 7.2-10.2c0-4.6-3-8.9-7.2-10.4L459.1 52.4zm-132.4 53c-12.5-12.5-32.8-12.5-45.3 0l-2.9 2.9C256.5 100.3 232.7 96 208 96C93.1 96 0 189.1 0 304S93.1 512 208 512s208-93.1 208-208c0-24.7-4.3-48.5-12.2-70.5l2.9-2.9c12.5-12.5 12.5-32.8 0-45.3l-80-80zM200 192c-57.4 0-104 46.6-104 104v8c0 8.8-7.2 16-16 16s-16-7.2-16-16v-8c0-75.1 60.9-136 136-136h8c8.8 0 16 7.2 16 16s-7.2 16-16 16h-8z"/></svg> Missing source or supplementary materials from evaluated methods/papers

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#0a8754;overflow:visible;position:relative;"><path d="M323.8 34.8c-38.2-10.9-78.1 11.2-89 49.4l-5.7 20c-3.7 13-10.4 25-19.5 35l-51.3 56.4c-8.9 9.8-8.2 25 1.6 33.9s25 8.2 33.9-1.6l51.3-56.4c14.1-15.5 24.4-34 30.1-54.1l5.7-20c3.6-12.7 16.9-20.1 29.7-16.5s20.1 16.9 16.5 29.7l-5.7 20c-5.7 19.9-14.7 38.7-26.6 55.5c-5.2 7.3-5.8 16.9-1.7 24.9s12.3 13 21.3 13L448 224c8.8 0 16 7.2 16 16c0 6.8-4.3 12.7-10.4 15c-7.4 2.8-13 9-14.9 16.7s.1 15.8 5.3 21.7c2.5 2.8 4 6.5 4 10.6c0 7.8-5.6 14.3-13 15.7c-8.2 1.6-15.1 7.3-18 15.1s-1.6 16.7 3.6 23.3c2.1 2.7 3.4 6.1 3.4 9.9c0 6.7-4.2 12.6-10.2 14.9c-11.5 4.5-17.7 16.9-14.4 28.8c.4 1.3 .6 2.8 .6 4.3c0 8.8-7.2 16-16 16H286.5c-12.6 0-25-3.7-35.5-10.7l-61.7-41.1c-11-7.4-25.9-4.4-33.3 6.7s-4.4 25.9 6.7 33.3l61.7 41.1c18.4 12.3 40 18.8 62.1 18.8H384c34.7 0 62.9-27.6 64-62c14.6-11.7 24-29.7 24-50c0-4.5-.5-8.8-1.3-13c15.4-11.7 25.3-30.2 25.3-51c0-6.5-1-12.8-2.8-18.7C504.8 273.7 512 257.7 512 240c0-35.3-28.6-64-64-64l-92.3 0c4.7-10.4 8.7-21.2 11.8-32.2l5.7-20c10.9-38.2-11.2-78.1-49.4-89zM32 192c-17.7 0-32 14.3-32 32V448c0 17.7 14.3 32 32 32H96c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32H32z"/></svg> Feedback to bioinformatics or general research community

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#0a8754;overflow:visible;position:relative;"><path d="M323.8 34.8c-38.2-10.9-78.1 11.2-89 49.4l-5.7 20c-3.7 13-10.4 25-19.5 35l-51.3 56.4c-8.9 9.8-8.2 25 1.6 33.9s25 8.2 33.9-1.6l51.3-56.4c14.1-15.5 24.4-34 30.1-54.1l5.7-20c3.6-12.7 16.9-20.1 29.7-16.5s20.1 16.9 16.5 29.7l-5.7 20c-5.7 19.9-14.7 38.7-26.6 55.5c-5.2 7.3-5.8 16.9-1.7 24.9s12.3 13 21.3 13L448 224c8.8 0 16 7.2 16 16c0 6.8-4.3 12.7-10.4 15c-7.4 2.8-13 9-14.9 16.7s.1 15.8 5.3 21.7c2.5 2.8 4 6.5 4 10.6c0 7.8-5.6 14.3-13 15.7c-8.2 1.6-15.1 7.3-18 15.1s-1.6 16.7 3.6 23.3c2.1 2.7 3.4 6.1 3.4 9.9c0 6.7-4.2 12.6-10.2 14.9c-11.5 4.5-17.7 16.9-14.4 28.8c.4 1.3 .6 2.8 .6 4.3c0 8.8-7.2 16-16 16H286.5c-12.6 0-25-3.7-35.5-10.7l-61.7-41.1c-11-7.4-25.9-4.4-33.3 6.7s-4.4 25.9 6.7 33.3l61.7 41.1c18.4 12.3 40 18.8 62.1 18.8H384c34.7 0 62.9-27.6 64-62c14.6-11.7 24-29.7 24-50c0-4.5-.5-8.8-1.3-13c15.4-11.7 25.3-30.2 25.3-51c0-6.5-1-12.8-2.8-18.7C504.8 273.7 512 257.7 512 240c0-35.3-28.6-64-64-64l-92.3 0c4.7-10.4 8.7-21.2 11.8-32.2l5.7-20c10.9-38.2-11.2-78.1-49.4-89zM32 192c-17.7 0-32 14.3-32 32V448c0 17.7 14.3 32 32 32H96c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32H32z"/></svg> Extendable to other field that requires intensive computation

<svg aria-hidden="true" role="img" viewBox="0 0 512 512" style="height:1em;width:1em;vertical-align:-0.125em;margin-left:auto;margin-right:auto;font-size:inherit;fill:#0a8754;overflow:visible;position:relative;"><path d="M323.8 34.8c-38.2-10.9-78.1 11.2-89 49.4l-5.7 20c-3.7 13-10.4 25-19.5 35l-51.3 56.4c-8.9 9.8-8.2 25 1.6 33.9s25 8.2 33.9-1.6l51.3-56.4c14.1-15.5 24.4-34 30.1-54.1l5.7-20c3.6-12.7 16.9-20.1 29.7-16.5s20.1 16.9 16.5 29.7l-5.7 20c-5.7 19.9-14.7 38.7-26.6 55.5c-5.2 7.3-5.8 16.9-1.7 24.9s12.3 13 21.3 13L448 224c8.8 0 16 7.2 16 16c0 6.8-4.3 12.7-10.4 15c-7.4 2.8-13 9-14.9 16.7s.1 15.8 5.3 21.7c2.5 2.8 4 6.5 4 10.6c0 7.8-5.6 14.3-13 15.7c-8.2 1.6-15.1 7.3-18 15.1s-1.6 16.7 3.6 23.3c2.1 2.7 3.4 6.1 3.4 9.9c0 6.7-4.2 12.6-10.2 14.9c-11.5 4.5-17.7 16.9-14.4 28.8c.4 1.3 .6 2.8 .6 4.3c0 8.8-7.2 16-16 16H286.5c-12.6 0-25-3.7-35.5-10.7l-61.7-41.1c-11-7.4-25.9-4.4-33.3 6.7s-4.4 25.9 6.7 33.3l61.7 41.1c18.4 12.3 40 18.8 62.1 18.8H384c34.7 0 62.9-27.6 64-62c14.6-11.7 24-29.7 24-50c0-4.5-.5-8.8-1.3-13c15.4-11.7 25.3-30.2 25.3-51c0-6.5-1-12.8-2.8-18.7C504.8 273.7 512 257.7 512 240c0-35.3-28.6-64-64-64l-92.3 0c4.7-10.4 8.7-21.2 11.8-32.2l5.7-20c10.9-38.2-11.2-78.1-49.4-89zM32 192c-17.7 0-32 14.3-32 32V448c0 17.7 14.3 32 32 32H96c17.7 0 32-14.3 32-32V224c0-17.7-14.3-32-32-32H32z"/></svg> Save time from selecting reproducible ML methods for biological problem

.emphasis[

You don't need to understand tons of cs to use this project

]

---

## Conclusion

1. The best performing methods on the real-world data will provide insights into why certain methods are better than others with respect predictive performance and capturing the correct underlying biological processes. 

2. Inform the development of a new integrative method using .hand[logistic regression] that is applicable to different types of multiomics data.

3. Highly relevant to the community utilizing multiomics data for
their research.
  + Ease of use
  + Collaborations
  + Development of new tool

---

class: middle
background-image: url(materials/img/website_bkg.png)
background-position: top
background-size: contain

&lt;br&gt;
&lt;br&gt;
&lt;br&gt;

.large[.secondary[Thanks to my lab, the [CompBio Lab](https://cbl-hli.med.ubc.ca/)]]
.pull-left[
- Dr. Amrit Singh
- Dr. Dantong Zhu
- Dr. Young Woong Kim
- Dr. Maryam Ahmadzadeh
- Michael Yoon
]
.pull-right[
- Jeffrey Tang
- Rishika Daswani
- Roy He
- Samuel Leung
- Akshdeep Sandhu
- Asees Singh
]

&lt;br&gt;

And acknowledging to UBC ARC and Nextflow for enabling the successful execution of  this project.

.large[
.footnote[
For more information regarding Nextflow, please check [here](https://www.nextflow.io/)
]]



&lt;!----
Credits to:

  [Stat 406 Lecture Slides @DJM](https://ubc-stat.github.io/stat-406/)

&lt;br&gt;&lt;br&gt;
-----&gt;

    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script src="materials/macros.js"></script>
<script>var slideshow = remark.create({
"highlightLines": true,
"countIncrementalSlides": false,
"ratio": "16:9"
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
// add `data-at-shortcutkeys` attribute to <body> to resolve conflicts with JAWS
// screen reader (see PR #262)
(function(d) {
  let res = {};
  d.querySelectorAll('.remark-help-content table tr').forEach(tr => {
    const t = tr.querySelector('td:nth-child(2)').innerText;
    tr.querySelectorAll('td:first-child .key').forEach(key => {
      const k = key.innerText;
      if (/^[a-z]$/.test(k)) res[k] = t;  // must be a single letter (key)
    });
  });
  d.body.setAttribute('data-at-shortcutkeys', JSON.stringify(res));
})(document);
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
